### DISCUSSION TITLE
SSE

### DISCUSSION THREAD

**Discussion by S222em** [nodejs/undici#1352] [Q&A] [answered]
Hello,

Is there a way I can use undici for SSE (server side events)? If so, could I utilize one connection to send requests to an API, and receive events over SSE?

I am currently using https://github.com/EventSource/eventsource for this, but I would like to use undici.

**Comment by mcollina** [comment, marked as answer]
SSE is not supported atm, but it would be a great addition. 
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Reply by S222em** [reply]
I found that undici actually does support this, but not the `text/event-stream` parsing to JSON.

```ts
const { body, statusCode } = await request('some-route', {
	method: 'GET',
	headers: {
		Accept: 'text/event-stream',
	},
	bodyTimeout: 0,
	dispatcher: this.dispatcher,
});

if (statusCode === 200) {
	body.setEncoding('utf8');
	body.on('data', (raw) => this.onMessage(raw));
        body.on('close', () => this.onDisconnect());
}
```

**Comment by dglozic** [comment]
I can confirm! It does feel like `stream` would be even better because we would not have to manually pass message chunks from readable to writable, but I could not get it to work.

What I noticed though is that because my microservice sits behind nginx proxy (even in localhost), I had to mess with switches to get it to not chunk the stream:

```nginx
location /wx/api/streaming/generate_stream {
			proxy_pass https://127.0.0.1:8462/;
			proxy_http_version 1.1;
			proxy_set_header Host $host:$server_port;
			proxy_set_header X-Forwarded-For $remote_addr;
			proxy_set_header Connection "";
			chunked_transfer_encoding off;
			proxy_buffering off;
			proxy_cache off;
		}
```

I am guessing all those `off` switches are needed to not mess with `text/event-stream` token messages as they are flying in.
